::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
: Copyright (c) 2018 Enea AB and others.
:
: All rights reserved. This program and the accompanying materials
: are made available under the terms of the Apache License, Version 2.0
: which accompanies this distribution, and is available at
: http://www.apache.org/licenses/LICENSE-2.0
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Tue, 18 Jul 2017 19:17:24 +0200
Subject: [PATCH] network: public: Use arm-virtual2 POD config

grep -e "10\.16\.0\." -R . -l | \
  xargs sed -i \
    -e 's/10\.16\.0\./10.0.9./g' \
    -e 's/10\.0\.9\.254/10.0.9.200/g' \
    -e 's/10\.0\.9\.1/10.0.9.254/g'

NOTE: This should be converted into a dynamic configuration read from
the universal POD descriptor in pharos, once that is ready.

Until then, just align the public network configuration used by the
virtual POD with the Enea lab configuration specific to arm-virtual2
(i.e. public network on 10.0.9.0/24).

NOTE: Replace the gateway at 10.16.0.1 (now 10.0.9.1) with the same
IP address as our lab's gateway (10.0.9.254), to keep both possible
network layouts in sync (using all virtual networks created via
virsh, respectively our lab's static config).
This will ensure deploys continue to work in both enviroments.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 mcp/config/states/networks                                        | 8 ++++----
 .../classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml | 4 ++--
 .../cluster/virtual-mcp-pike-common-noha/openstack_init.yml       | 2 +-
 .../classes/cluster/virtual-mcp-pike-odl-noha/infra/config.yml    | 2 +-
 .../cluster/virtual-mcp-pike-ovs-dpdk-noha/infra/config.yml       | 2 +-
 .../classes/cluster/virtual-mcp-pike-ovs-noha/infra/config.yml    | 2 +-
 mcp/scripts/net_public.xml                                        | 2 +-
 7 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/mcp/config/states/networks b/mcp/config/states/networks
index 222a072..9bbc0ce 100755
--- a/mcp/config/states/networks
+++ b/mcp/config/states/networks
@@ -17,10 +17,10 @@ POOL_START_IP=$(get_nova_compute_pillar_data 'opnfv_net_public_pool_start')
 POOL_END_IP=$(get_nova_compute_pillar_data 'opnfv_net_public_pool_end')

 # Default values
-PUBLIC_NET="${PUBLIC_NET:-10.16.0.0/24}"
-PUBLIC_NET_GATEWAY="${PUBLIC_NET_GATEWAY:-10.16.0.1}"
-POOL_START_IP="${POOL_START_IP:-10.16.0.130}"
-POOL_END_IP="${POOL_END_IP:-10.16.0.254}"
+PUBLIC_NET="${PUBLIC_NET:-10.0.9.0/24}"
+PUBLIC_NET_GATEWAY="${PUBLIC_NET_GATEWAY:-10.0.9.254}"
+POOL_START_IP="${POOL_START_IP:-10.0.9.130}"
+POOL_END_IP="${POOL_END_IP:-10.0.9.200}"

 # Print openstack status and setup network
 salt -C 'I@nova:controller and *01*' cmd.run ". /root/keystonercv3; \
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
index 7e3a494..3a240b0 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/infra/config.yml
@@ -61,9 +61,9 @@ parameters:
           params:
             single_address: 172.16.10.105
             tenant_address: 10.1.0.105
-            external_address: 10.16.0.105
+            external_address: 10.0.9.105
         openstack_compute_node02:
           params:
             single_address: 172.16.10.106
             tenant_address: 10.1.0.106
-            external_address: 10.16.0.106
+            external_address: 10.0.9.106
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
index d5606b5..508d5e1 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-common-noha/openstack_init.yml
@@ -14,7 +14,7 @@ parameters:
     openstack_region: RegionOne
     admin_email: root@localhost
     cluster_public_protocol: http
-    cluster_public_host: 10.16.0.101
+    cluster_public_host: 10.0.9.101
     neutron_public_protocol: http
     neutron_control_dvr: 'False'
     neutron_l3_ha: 'False'
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-odl-noha/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-odl-noha/infra/config.yml
index 8e38ff8..d7bf017 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-odl-noha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-odl-noha/infra/config.yml
@@ -24,4 +24,4 @@ parameters:
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-dpdk-noha/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-dpdk-noha/infra/config.yml
index e80c442..1df99b4 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-dpdk-noha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-dpdk-noha/infra/config.yml
@@ -25,4 +25,4 @@ parameters:
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-noha/infra/config.yml b/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-noha/infra/config.yml
index 66c2468..7326295 100644
--- a/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-noha/infra/config.yml
+++ b/mcp/reclass/classes/cluster/virtual-mcp-pike-ovs-noha/infra/config.yml
@@ -17,4 +17,4 @@ parameters:
         openstack_gateway_node01:
           params:
             tenant_address: 10.1.0.110
-            external_address: 10.16.0.110
+            external_address: 10.0.9.110
diff --git a/mcp/scripts/net_public.xml b/mcp/scripts/net_public.xml
index d6df4aa..87f3146 100644
--- a/mcp/scripts/net_public.xml
+++ b/mcp/scripts/net_public.xml
@@ -10,5 +10,5 @@
   <name>public</name>
   <bridge name="public"/>
   <forward mode="nat"/>
-  <ip address="10.16.0.1" netmask="255.255.255.0" />
+  <ip address="10.0.9.254" netmask="255.255.255.0" />
 </network>
